diff -Naur -x '*.svn*' -x '*.mo' -x '*.pyc' -x '._*' mod_wsgi-8994b058d2db.org/src/ngx_http_wsgi_handler.c mod_wsgi-8994b058d2db/src/ngx_http_wsgi_handler.c
--- mod_wsgi-8994b058d2db.org/src/ngx_http_wsgi_handler.c	2008-03-26 20:35:15.000000000 +0100
+++ mod_wsgi-8994b058d2db/src/ngx_http_wsgi_handler.c	2009-02-09 04:22:38.000000000 +0100
@@ -71,7 +71,7 @@
 
   if (r->method == NGX_HTTP_GET || r->method == NGX_HTTP_HEAD) {
     /* XXX not sure */
-    rc = ngx_http_discard_body(r);
+    rc = ngx_http_discard_request_body(r);
 
     if (rc != NGX_OK && rc != NGX_AGAIN) {
       return rc;
diff -Naur -x '*.svn*' -x '*.mo' -x '*.pyc' -x '._*' mod_wsgi-8994b058d2db.org/src/ngx_wsgi.c mod_wsgi-8994b058d2db/src/ngx_wsgi.c
--- mod_wsgi-8994b058d2db.org/src/ngx_wsgi.c	2008-03-26 20:35:15.000000000 +0100
+++ mod_wsgi-8994b058d2db/src/ngx_wsgi.c	2009-02-09 04:21:36.000000000 +0100
@@ -1545,7 +1545,7 @@
 
     if (PyTuple_Size(tuple) != 2) {
       PyErr_Format(PyExc_ValueError, "tuple of length 2 "
-                   "expected, length is %zd",
+                   "expected, length is %d",
                    PyTuple_Size(tuple));
       return NGX_ERROR;
     }
