diff -Naur sphinx-0.9.8-rc2.org/Makefile.am sphinx-0.9.8-rc2/Makefile.am
--- sphinx-0.9.8-rc2.org/Makefile.am	2008-03-28 17:50:12.000000000 +0100
+++ sphinx-0.9.8-rc2/Makefile.am	2008-03-28 17:50:36.000000000 +0100
@@ -5,7 +5,7 @@
 endif
 
 EXTRA_DIST = api storage sphinx.conf.in example.sql
-sysconf_DATA = sphinx.conf.dist example.sql
+#sysconf_DATA = sphinx.conf.dist example.sql
 
 install-data-hook:
-	mkdir -p $(DESTDIR)$(localstatedir)/data && mkdir -p $(DESTDIR)$(localstatedir)/log
+#	mkdir -p $(DESTDIR)$(localstatedir)/data && mkdir -p $(DESTDIR)$(localstatedir)/log
